examples/event_handling/lasso_demo.py

```diff
@@ -84,6 +86,10 @@ def on_press(self, event):
         # acquire a lock on the widget drawing
         self.canvas.widgetlock(self.lasso)
 
+    def on_release(self, event):
+        if self.canvas.widgetlock.isowner(self.lasso):
+            self.canvas.widgetlock.release(self.lasso)
```

Comment:
If the widget lock is already taken in `on_press`, then it would return and not set `self.lasso`. So you cannot use `self.lasso` here directly or you will get an `AttributeError`. This needs to check
```suggestion
        if hasattr(self, 'lasso') and self.canvas.widgetlock.isowner(self.lasso):
            self.canvas.widgetlock.release(self.lasso)
```